src/select/cmp.rs

```diff
@@ -0,0 +1,361 @@
+use crate::select::select_value::SelectValue;
+use array_tool::vec::{Intersect, Union};
```

Comment:
I'm not sure it's a good idea to add the `array_tool` crate just for this. It also looks old and unmaintained.
The intersection/union can be done with the standard `HashSet` type:
https://doc.rust-lang.org/std/collections/hash_set/index.html

Reply:
Removed